### PR TITLE
PowerShell co-ordinated build OneBranch pipeline

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

One branch pipeline for building source code for PowerShell on various Windows. Linux and macOS platforms.

## PR Context

Convert current pipelines to OneBranch

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [x] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: .config/suppress.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (TravisEz13) [inline .config/suppress.json:1]: Do we need both files

File: .pipelines/PowerShell-Coordinated_Packages-Official.yml [hunk 2]
```diff
@@ -0,0 +1,270 @@
+#################################################################################
+#                               OneBranch Pipelines                             #
+# This pipeline was created by EasyStart from a sample located at:              #
+#   https://aka.ms/obpipelines/easystart/samples                                #
+# Documentation:  https://aka.ms/obpipelines                                    #
+# Yaml Schema:    https://aka.ms/obpipelines/yaml/schema                        #
+# Retail Tasks:   https://aka.ms/obpipelines/tasks                              #
+# Support:        https://aka.ms/onebranchsup                                   #
```
Comment (TravisEz13) [inline .pipelines/PowerShell-Coordinated_Packages-Official.yml:8]:
```suggestion
```

File: .pipelines/PowerShell-Coordinated_Packages-Official.yml [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+      #   parameters:
+      #     buildName: rpm
+      #     parentJob: build_deb
+      # - template: /.pipelines/templates/linux.yml@self
+      #   parameters:
+      #     buildName: fxdependent
+      #     parentJob: build_deb
+      # - template: /.pipelines/templates/linux.yml@self
+      #   parameters:
+      #     buildName: alpine
```
Comment (TravisEz13) [inline .pipelines/PowerShell-Coordinated_Packages-Official.yml:204]:
```suggestion
```

File: .pipelines/PowerShell-Coordinated_Packages-Official.yml [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+  - name: branchCounterKey
+    value: $[format('{0:yyyyMMdd}-{1}', pipeline.startTime,variables['Build.SourceBranch'])]
+  - name: branchCounter
+    value: $[counter(variables['branchCounterKey'], 1)]
+  - name: ForceAzureBlobDelete
+    value: ${{ parameters.ForceAzureBlobDelete }}
+  - name: Github_Build_Repository_Uri
+    value: https://github.com/powershell/powershell
+  - name: SBOMGenerator_Formats
+    value: spdx:2.2
```
Comment (TravisEz13) [inline .pipelines/PowerShell-Coordinated_Packages-Official.yml:78]:
```suggestion
```

File: .pipelines/PowerShell-Coordinated_Packages-Official.yml [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+        scanFolder:  $(Build.SourcesDirectory)
+        suppressionsFile: $(Build.SourcesDirectory)\.config\suppress.json
+      asyncSdl: # https://aka.ms/obpipelines/asyncsdl
+        enabled: true
+        forStages: [prep, macos, linux, windows, SignFiles, test_and_release_artifacts]
+        credscan:
+          enabled: true
+          scanFolder:  $(Build.SourcesDirectory)
+          suppressionsFile: $(Build.SourcesDirectory)\PowerShell\.config\suppress.json
+        binskim:
```
Comment (TravisEz13) [inline .pipelines/PowerShell-Coordinated_Packages-Official.yml:123]: Add a comment explaining why it is disabled

File: .pipelines/PowerShell-Coordinated_Packages-Official.yml [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+  - group: 'AzDevOpsArtifacts'
+
+extends:
+  template: v2/OneBranch.Official.CrossPlat.yml@onebranchTemplates
+  parameters:
+    customTags: 'ES365AIMigrationTooling'
+    globalSdl:
+      disableLegacyManifest: true
+      armory:
+        enabled: false
```
Comment (TravisEz13) [inline .pipelines/PowerShell-Coordinated_Packages-Official.yml:103]: Add a comment explaining why it is disabled

File: .pipelines/PowerShell-Coordinated_Packages-Official.yml [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+        enabled: true
+        forStages: [prep, macos, linux, windows, SignFiles, test_and_release_artifacts]
+        credscan:
+          enabled: true
+          scanFolder:  $(Build.SourcesDirectory)
+          suppressionsFile: $(Build.SourcesDirectory)\PowerShell\.config\suppress.json
+        binskim:
+          enabled: false
+        apiscan:
+          enabled: false
```
Comment (TravisEz13) [inline .pipelines/PowerShell-Coordinated_Packages-Official.yml:126]:
```suggestion
          # APIScan requires a non-Ready-To-Run build
          enabled: false
```

File: .pipelines/PowerShell-Coordinated_Packages-Official.yml [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+      #     buildName: alpine
+    - stage: windows
+      dependsOn: ['prep']
+      jobs:
+      - template: /.pipelines/templates/windows-hosted-build.yml@self
+        parameters:
+          Architecture: x64
+          BuildConfiguration: release
+          JobName: build_windows_x64_release
+      - template: /.pipelines/templates/windows-hosted-build.yml@self
```
Comment (TravisEz13) [inline .pipelines/PowerShell-Coordinated_Packages-Official.yml:213]:
```suggestion

      - template: /.pipelines/templates/windows-hosted-build.yml@self
```
Can we add a new line between jobs?  I find it a lot easier to read.

File: .pipelines/templates/SetVersionVariables.yml [hunk 9]
```diff
@@ -0,0 +1,70 @@
+# This Yaml Document has been converted by ESAI Yaml Pipeline Conversion Tool.
+# Please make sure to check all the converted content, it is your team's responsibility to make sure that the pipeline is still valid and functions as expected.
```
Comment (TravisEz13) [inline .pipelines/templates/SetVersionVariables.yml:2]:
```suggestion
```

File: .pipelines/templates/compliance.yml [hunk 10]
```diff
@@ -0,0 +1,46 @@
+# This Yaml Document has been converted by ESAI Yaml Pipeline Conversion Tool.
+# Please make sure to check all the converted content, it is your team's responsibility to make sure that the pipeline is still valid and functions as expected.
```
Comment (TravisEz13) [inline .pipelines/templates/compliance.yml:2]:
```suggestion
```

File: .pipelines/templates/expand-compliance.yml [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+parameters:
+  architecture: x86
+  version: 6.2.0
+steps:
+- powershell: |
+    Expand-Archive -Path "$(System.ArtifactsDirectory)\results\PowerShell-${{ parameters.version }}-symbols-win-${{ parameters.architecture }}.zip" -Destination "$(Build.StagingDirectory)\symbols\${{ parameters.architecture }}"
+  displayName: Expand symbols zip - ${{ parameters.architecture }}
+- powershell: |
+    tools/releaseBuild/createComplianceFolder.ps1 -ArtifactFolder "$(Build.StagingDirectory)\symbols\${{ parameters.architecture }}" -VSTSVariableName 'CompliancePath'
+  displayName: Expand Compliance file - ${{ parameters.architecture }}
```
Comment (TravisEz13) [inline .pipelines/templates/expand-compliance.yml:12]: no posix file ending

Comment (TravisEz13) [inline .pipelines/templates/expand-compliance.yml:12]:
```suggestion
  displayName: Expand Compliance file - ${{ parameters.architecture }}

```
add posix file ending

File: .pipelines/templates/global-tool-pkg-sbom.yml [hunk 12]
```diff
@@ -0,0 +1,54 @@
+# This Yaml Document has been converted by ESAI Yaml Pipeline Conversion Tool.
+# Please make sure to check all the converted content, it is your team's responsibility to make sure that the pipeline is still valid and functions as expected.
```
Comment (TravisEz13) [inline .pipelines/templates/global-tool-pkg-sbom.yml:2]:
```suggestion
```

File: .pipelines/templates/global-tool-pkg-sbom.yml [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+      Build_Repository_Uri: 'https://github.com/powershell/powershell'
+      PackageName: $(GlobalToolPkgName)
+      PackageVersion: ${{ parameters.PackageVersion }}
+      sourceScanPath: $(GlobalToolCGManifestPath)
+      displayName: SBOM for Global Tool package
+  - pwsh: |
+      Import-Module -Name $env:REPOROOT\build.psm1
+      Import-Module -Name $env:REPOROOT\tools\packaging
+      New-GlobalToolNupkgFromSource -PackageNuSpecPath "$env:GlobalToolNuSpecSourcePath" -PackageName "$env:GlobalToolPkgName" -DestinationPath '${{ parameters.DestinationPath }}' -CGManifestPath "$env:GlobalToolCGManifestPath"
+    displayName: 'Create global tool NuSpec package from NuSpec source.'
```
Comment (TravisEz13) [inline .pipelines/templates/global-tool-pkg-sbom.yml:54]:
```suggestion
    displayName: 'Create global tool NuSpec package from NuSpec source.'

```

File: tools/releaseBuild/setReleaseTag.ps1 [hunk 14]
```diff
@@ -49,6 +49,12 @@ function New-BuildInfoJson {
     Write-Host -Object "##$vstsCommandString"
 
     Write-Host "##vso[artifact.upload containerfolder=BuildInfoJson;artifactname=BuildInfoJson]$resolvedPath"
+
+    if (-not (Test-Path $env:ob_outputDirectory)) {
+        $null = New-Item -Path $env:ob_outputDirectory -ItemType Directory -Force -Verbose
+    }
+
+    Copy-Item $resolvedPath -Destination $env:ob_outputDirectory -Force -Verbose
```
Comment (TravisEz13) [inline tools/releaseBuild/setReleaseTag.ps1:57]:
```suggestion

    if (-not (Test-Path $env:ob_outputDirectory)) {
        $null = New-Item -Path $env:ob_outputDirectory -ItemType Directory -Force -Verbose
    }
    
    # Upload for ADO pipelines
    Write-Host "##vso[artifact.upload containerfolder=BuildInfoJson;artifactname=BuildInfoJson]$resolvedPath"
    # Copy to location where OneBranch Pipelines uploads from
    Copy-Item $resolvedPath -Destination $env:ob_outputDirectory -Force -Verbose
```

File: .config/suppress.json (hunk 1 above)
Comment (adityapatwardhan) [reply .config/suppress.json:1]: Removed the csk

File: .config/tsaoptions.json [hunk 15]
```diff
@@ -0,0 +1,12 @@
...
+    "instanceUrl": "https://msazure.visualstudio.com",
+    "projectName": "One",
+    "areaPath": "One\\MGMT\\Compute\\Powershell\\Powershell\\PowerShell Core",
+    "notificationAliases": [
+        "adityap@microsoft.com",
+        "tplunk@microsoft.com",
+        "dongbow@microsoft.com",
+        "pmeinecke@microsoft.com"
+    ],
+    "codebaseName": "PowerShellCore_202403"
```
[the remark this answers is not in the document]
Comment (adityapatwardhan) [reply .config/tsaoptions.json:11]: Removed

File: .pipelines/PowerShell-Coordinated_Packages-Official.yml (hunk 2 above)
Comment (adityapatwardhan) [reply .pipelines/PowerShell-Coordinated_Packages-Official.yml:8]: Removed

File: .pipelines/PowerShell-Coordinated_Packages-Official.yml (hunk 4 above)
Comment (adityapatwardhan) [reply .pipelines/PowerShell-Coordinated_Packages-Official.yml:78]: Removed

File: .pipelines/PowerShell-Coordinated_Packages-Official.yml (hunk 5 above)
Comment (adityapatwardhan) [reply .pipelines/PowerShell-Coordinated_Packages-Official.yml:123]: Added comment

File: .pipelines/PowerShell-Coordinated_Packages-Official.yml (hunk 7 above)
Comment (adityapatwardhan) [reply .pipelines/PowerShell-Coordinated_Packages-Official.yml:126]: added

File: .pipelines/PowerShell-Coordinated_Packages-Official.yml (hunk 3 above)
Comment (adityapatwardhan) [reply .pipelines/PowerShell-Coordinated_Packages-Official.yml:204]: removed

File: .pipelines/PowerShell-Coordinated_Packages-Official.yml (hunk 8 above)
Comment (adityapatwardhan) [reply .pipelines/PowerShell-Coordinated_Packages-Official.yml:213]: done

File: .pipelines/templates/SetVersionVariables.yml (hunk 9 above)
Comment (adityapatwardhan) [reply .pipelines/templates/SetVersionVariables.yml:2]: removed

File: .pipelines/templates/cloneToOfficialPath.yml [hunk 16]
```diff
@@ -0,0 +1,21 @@
+# This Yaml Document has been converted by ESAI Yaml Pipeline Conversion Tool.
+# Please make sure to check all the converted content, it is your team's responsibility to make sure that the pipeline is still valid and functions as expected.
```
[the remark this answers is not in the document]
Comment (adityapatwardhan) [reply .pipelines/templates/cloneToOfficialPath.yml:2]: removed

File: .pipelines/templates/compliance.yml (hunk 10 above)
Comment (adityapatwardhan) [reply .pipelines/templates/compliance.yml:2]: removed

File: .pipelines/templates/expand-compliance.yml (hunk 11 above)
Comment (adityapatwardhan) [reply .pipelines/templates/expand-compliance.yml:12]: fixed

File: .pipelines/templates/global-tool-pkg-sbom.yml (hunk 12 above)
Comment (adityapatwardhan) [reply .pipelines/templates/global-tool-pkg-sbom.yml:2]: removed

File: .pipelines/templates/global-tool-pkg-sbom.yml (hunk 13 above)
Comment (adityapatwardhan) [reply .pipelines/templates/global-tool-pkg-sbom.yml:54]: Added

File: tools/releaseBuild/setReleaseTag.ps1 (hunk 14 above)
Comment (adityapatwardhan) [reply tools/releaseBuild/setReleaseTag.ps1:57]: fixed

File: .pipelines/templates/checkAzureContainer.yml [hunk 17]
```diff
@@ -0,0 +1,92 @@
+# This Yaml Document has been converted by ESAI Yaml Pipeline Conversion Tool.
+# Please make sure to check all the converted content, it is your team's responsibility to make sure that the pipeline is still valid and functions as expected.
```
Comment (TravisEz13) [inline .pipelines/templates/checkAzureContainer.yml:2]:
```suggestion
```

File: .pipelines/templates/compliance.yml [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+    continueOnError: true
+  - template: /.pipelines/templates/expand-compliance.yml@self
+    parameters:
+      architecture: fxdependent
+      version: $(version)
+  - template: /.pipelines/templates/expand-compliance.yml@self
+    parameters:
+      architecture: x86
+      version: $(version)
+  - template: /.pipelines/templates/expand-compliance.yml@self
```
Comment (TravisEz13) [inline .pipelines/templates/compliance.yml:43]: does this do anything?  Is it supposed to?

File: .pipelines/templates/expand-compliance.yml [hunk 19]
```diff
@@ -0,0 +1,12 @@
+# This Yaml Document has been converted by ESAI Yaml Pipeline Conversion Tool.
+# Please make sure to check all the converted content, it is your team's responsibility to make sure that the pipeline is still valid and functions as expected.
```
Comment (TravisEz13) [inline .pipelines/templates/expand-compliance.yml:2]:
```suggestion
```

File: .pipelines/templates/insert-nuget-config-azfeed.yml [hunk 20]
```diff
@@ -0,0 +1,35 @@
+# This Yaml Document has been converted by ESAI Yaml Pipeline Conversion Tool.
+# Please make sure to check all the converted content, it is your team's responsibility to make sure that the pipeline is still valid and functions as expected.
```
Comment (TravisEz13) [inline .pipelines/templates/insert-nuget-config-azfeed.yml:2]:
```suggestion
```

File: .pipelines/templates/insert-nuget-config-azfeed.yml [hunk 21]
```diff
@@ -0,0 +1,35 @@
...
+    if(-not (Test-Path $configPath))
+    {
+        throw "nuget.config is not created"
+    }
+    Get-Content $configPath | Write-Verbose -Verbose
+  displayName: 'Add nuget.config for Azure DevOps feed for packages'
+  condition: and(succeededOrFailed(), ne(variables['PSInternalNugetFeed'], ''))
+  env:
+    NugetConfigDir: ${{ parameters.repoRoot }}
+    ob_restore_phase: true # This ensures checkout is done at the beginning of the restore phase
```
Comment (TravisEz13) [inline .pipelines/templates/insert-nuget-config-azfeed.yml:35]: Do we use nuget.org, if so, we need to replace it too.

I'll send you the reference offline.

File: .pipelines/templates/nuget-pkg-sbom.yml [hunk 22]
```diff
@@ -0,0 +1,133 @@
...
+      $FileName = '${{ value }}'
+      $FileBaseName = [System.IO.Path]::GetFileNameWithoutExtension($FileName)
+      $FilePackagePath = Join-Path -Path '${{ parameters.PackagePath }}' -ChildPath $FileBaseName
+      Write-Verbose -Verbose "FileName to package: $FileName"
+      Write-Verbose -Verbose "FilePackage path: $FilePackagePath"
+      Import-Module -Name $env:REPOROOT\build.psm1
+      Import-Module -Name $env:REPOROOT\tools\packaging
+      Find-DotNet
+      New-ILNugetPackageFromSource -FileName $FileName -PackageVersion '${{ parameters.PackageVersion }}' -PackagePath '${{ parameters.PackagePath }}'
+    displayName: 'Create NuGet Package for single file'
```
Comment (TravisEz13) [inline .pipelines/templates/nuget-pkg-sbom.yml:133]:
```suggestion
    displayName: 'Create NuGet Package for single file'

```

File: .pipelines/templates/shouldSign.yml [hunk 23]
```diff
@@ -0,0 +1,25 @@
...
+    $vstsCommandString = "vso[task.setvariable variable=SHOULD_SIGN]$($shouldSign.ToString().ToLowerInvariant())"
+    Write-Host "sending " + $vstsCommandString
+    Write-Host "##$vstsCommandString"
+    $vstsCommandString = "vso[task.setvariable variable=MSIX_CERT]$($msixCert)"
+    Write-Host "sending " + $vstsCommandString
+    Write-Host "##$vstsCommandString"
+    $vstsCommandString = "vso[task.setvariable variable=AUTHENTICODE_CERT]$($authenticodeCert)"
+    Write-Host "sending " + $vstsCommandString
+    Write-Host "##$vstsCommandString"
+  displayName: 'Set SHOULD_SIGN Variable'
```
Comment (TravisEz13) [inline .pipelines/templates/shouldSign.yml:25]:
```suggestion
  displayName: 'Set SHOULD_SIGN Variable'

```

File: .pipelines/templates/step/finalize.yml [hunk 24]
```diff
@@ -0,0 +1,7 @@
+# This Yaml Document has been converted by ESAI Yaml Pipeline Conversion Tool.
+# Please make sure to check all the converted content, it is your team's responsibility to make sure that the pipeline is still valid and functions as expected.
+steps:
+- pwsh: |
```
Comment (TravisEz13) [inline .pipelines/templates/step/finalize.yml:4]: This may not be needed in OneBranch

File: .pipelines/templates/upload-final-results.yml [hunk 25]
```diff
@@ -0,0 +1,17 @@
...
+steps:
+- powershell: |
+    Get-ChildItem -Path '${{ parameters.artifactPath }}' -Recurse -File -filter '${{ parameters.artifactFilter }}' -ErrorAction SilentlyContinue |
+        Select-Object -ExpandProperty FullName |
+            ForEach-Object {
+                Write-Host "##vso[artifact.upload containerfolder=${{ parameters.artifactName }};artifactname=${{ parameters.artifactName }}]$_"
+            }
+  displayName: Upload ${{ parameters.artifactName }} Artifacts ${{ parameters.artifactFilter }} from ${{ parameters.artifactPath }}
+  condition: ${{ parameters.condition }}
+  retryCountOnTaskFailure: 2
```
Comment (TravisEz13) [inline .pipelines/templates/upload-final-results.yml:17]:
```suggestion
  retryCountOnTaskFailure: 2

```

File: .pipelines/PowerShell-Coordinated_Packages-Official.yml (hunk 6 above)
Comment (adityapatwardhan) [reply .pipelines/PowerShell-Coordinated_Packages-Official.yml:92]: added

File: .pipelines/templates/expand-compliance.yml (hunk 19 above)
Comment (adityapatwardhan) [reply .pipelines/templates/expand-compliance.yml]: removed

File: .pipelines/templates/shouldSign.yml (hunk 23 above)
Comment (adityapatwardhan) [reply .pipelines/templates/shouldSign.yml:25]: These files are going to be replaced with packaging changes

File: .pipelines/templates/step/finalize.yml (hunk 24 above)
Comment (adityapatwardhan) [reply .pipelines/templates/step/finalize.yml:4]: I will remove this with packaging changes

File: .pipelines/templates/upload-final-results.yml (hunk 25 above)
Comment (adityapatwardhan) [reply .pipelines/templates/upload-final-results.yml:17]: This file will be removed with packaging changes.

File: .pipelines/templates/nuget-pkg-sbom.yml (hunk 22 above)
Comment (adityapatwardhan) [reply .pipelines/templates/nuget-pkg-sbom.yml:133]: These files are going to be replaced with packaging changes

File: .pipelines/templates/compliance.yml (hunk 18 above)
Comment (adityapatwardhan) [reply .pipelines/templates/compliance.yml:41]: These files are going to be replaced with packaging changes

File: .pipelines/templates/insert-nuget-config-azfeed.yml (hunk 20 above)
Comment (adityapatwardhan) [reply .pipelines/templates/insert-nuget-config-azfeed.yml:2]: removed

File: .pipelines/templates/checkAzureContainer.yml (hunk 17 above)
Comment (adityapatwardhan) [reply .pipelines/templates/checkAzureContainer.yml:2]: removed

File: .pipelines/templates/compliance.yml [hunk 26]
```diff
@@ -0,0 +1,44 @@
+parameters:
```
Comment (adityapatwardhan) [inline .pipelines/templates/compliance.yml:1]: Not used anymore

File: .pipelines/templates/expand-compliance.yml [hunk 27]
```diff
@@ -0,0 +1,10 @@
+parameters:
```
Comment (adityapatwardhan) [inline .pipelines/templates/expand-compliance.yml:1]: not used anymore

File: .pipelines/templates/linux-packaging.yml [hunk 28]
```diff
@@ -0,0 +1,418 @@
+# This Yaml Document has been converted by ESAI Yaml Pipeline Conversion Tool.
```
Comment (adityapatwardhan) [inline .pipelines/templates/linux-packaging.yml:1]: not used anymore

File: .pipelines/templates/mac-file-signing.yml [hunk 29]
```diff
@@ -0,0 +1,96 @@
+# This Yaml Document has been converted by ESAI Yaml Pipeline Conversion Tool.
```
Comment (adityapatwardhan) [inline .pipelines/templates/mac-file-signing.yml:1]: not used anymore

File: .pipelines/templates/mac-package-build.yml [hunk 30]
```diff
@@ -0,0 +1,117 @@
+# This Yaml Document has been converted by ESAI Yaml Pipeline Conversion Tool.
```
Comment (adityapatwardhan) [inline .pipelines/templates/mac-package-build.yml:1]: not used anymore

File: .pipelines/templates/mac-package-signing.yml [hunk 31]
```diff
@@ -0,0 +1,113 @@
+# This Yaml Document has been converted by ESAI Yaml Pipeline Conversion Tool.
```
Comment (adityapatwardhan) [inline .pipelines/templates/mac-package-signing.yml:1]: not used anymore

File: .pipelines/templates/nuget.yml [hunk 32]
```diff
@@ -0,0 +1,228 @@
+# This Yaml Document has been converted by ESAI Yaml Pipeline Conversion Tool.
```
Comment (adityapatwardhan) [inline .pipelines/templates/nuget.yml:1]: not used anymore

File: .pipelines/templates/sign-build-file.yml [hunk 33]
```diff
@@ -0,0 +1,261 @@
+# This Yaml Document has been converted by ESAI Yaml Pipeline Conversion Tool.
```
Comment (adityapatwardhan) [inline .pipelines/templates/sign-build-file.yml:1]: not used anymore

File: .pipelines/templates/upload-final-results.yml [hunk 34]
```diff
@@ -0,0 +1,17 @@
+# This Yaml Document has been converted by ESAI Yaml Pipeline Conversion Tool.
```
Comment (adityapatwardhan) [inline .pipelines/templates/upload-final-results.yml:1]: not used anymore

File: .pipelines/templates/upload.yml [hunk 35]
```diff
@@ -0,0 +1,69 @@
+# This Yaml Document has been converted by ESAI Yaml Pipeline Conversion Tool.
```
Comment (adityapatwardhan) [inline .pipelines/templates/upload.yml:1]: not used anymore

File: .pipelines/templates/windows-packaging.yml [hunk 36]
```diff
@@ -0,0 +1,308 @@
+# This Yaml Document has been converted by ESAI Yaml Pipeline Conversion Tool.
+# Please make sure to check all the converted content, it is your team's responsibility to make sure that the pipeline is still valid and functions as expected.
+parameters:
```
Comment (adityapatwardhan) [inline .pipelines/templates/windows-packaging.yml:3]: not used anymore

File: .config/tsaoptions.json [hunk 37]
```diff
@@ -0,0 +1,11 @@
+{
+    "instanceUrl": "https://msazure.visualstudio.com",
+    "projectName": "One",
+    "areaPath": "One\\MGMT\\Compute\\Powershell\\Powershell\\PowerShell Core",
+    "notificationAliases": [
```
Comment (TravisEz13) [inline .config/tsaoptions.json:5]: see link sent offline

File: .pipelines/PowerShell-Coordinated_Packages-Official.yml [hunk 38]
```diff
@@ -0,0 +1,252 @@
...
+  - name: DOTNET_CLI_TELEMETRY_OPTOUT
+    value: 1
+  - name: POWERSHELL_TELEMETRY_OPTOUT
+    value: 1
+  - name: nugetMultiFeedWarnLevel
+    value: none
+  - name: NugetSecurityAnalysisWarningLevel
+    value: none
+  - name: skipNugetSecurityAnalysis
+    value: true
```
Comment (TravisEz13) [inline .pipelines/PowerShell-Coordinated_Packages-Official.yml:58]: In a new PR, we should try to remove these by using the restore phase.

File: .pipelines/PowerShell-Coordinated_Packages-Official.yml [hunk 39]
```diff
@@ -0,0 +1,252 @@
...
+    value: $[format('{0:yyyyMMdd}-{1}', pipeline.startTime,variables['Build.SourceBranch'])]
+  - name: branchCounter
+    value: $[counter(variables['branchCounterKey'], 1)]
+  - name: ForceAzureBlobDelete
+    value: ${{ parameters.ForceAzureBlobDelete }}
+  - name: Github_Build_Repository_Uri
+    value: https://github.com/powershell/powershell
+  - name: BUILDSECMON_OPT_IN
+    value: true
+  - group: PoolNames
```
Comment (TravisEz13) [inline .pipelines/PowerShell-Coordinated_Packages-Official.yml:69]: This shouldn't be needed.

File: .pipelines/PowerShell-Coordinated_Packages-Official.yml [hunk 40]
```diff
@@ -0,0 +1,252 @@
...
+    value: none
+  - name: skipNugetSecurityAnalysis
+    value: true
+  - name: branchCounterKey
+    value: $[format('{0:yyyyMMdd}-{1}', pipeline.startTime,variables['Build.SourceBranch'])]
+  - name: branchCounter
+    value: $[counter(variables['branchCounterKey'], 1)]
+  - name: ForceAzureBlobDelete
+    value: ${{ parameters.ForceAzureBlobDelete }}
+  - name: Github_Build_Repository_Uri
```
Comment (TravisEz13) [inline .pipelines/PowerShell-Coordinated_Packages-Official.yml:65]: not needed

File: .pipelines/PowerShell-Coordinated_Packages-Official.yml [hunk 41]
```diff
@@ -0,0 +1,252 @@
...
+  parameters:
+    customTags: 'ES365AIMigrationTooling'
+    globalSdl:
+      disableLegacyManifest: true
+      # disabled Armorty as we dont have any ARM templates to scan. It fails on some sample ARM templates.
+      armory:
+        enabled: false
+      sbom:
+        enabled: true
+        packageName: Microsoft.PowerShell
```
Comment (TravisEz13) [inline .pipelines/PowerShell-Coordinated_Packages-Official.yml:95]: set it in each build using the variable in each job:

```
        packageName: Microsoft.PowerShell.<Arch>
```

```suggestion
```

File: .pipelines/templates/cloneToOfficialPath.yml [hunk 42]
```diff
@@ -0,0 +1,19 @@
+parameters:
+  nativePathRoot: ''
+
+steps:
+- powershell: |
+    $dirSeparatorChar = [system.io.path]::DirectorySeparatorChar
```
Comment (TravisEz13) [inline .pipelines/templates/cloneToOfficialPath.yml:6]: We should try to remove this, but we can do that in a later PR.

File: .pipelines/templates/insert-nuget-config-azfeed.yml (hunk 20 above)
Comment (TravisEz13) [reply .pipelines/templates/insert-nuget-config-azfeed.yml:2]: still not resolved

File: .pipelines/templates/json.yml [hunk 43]
```diff
@@ -0,0 +1,43 @@
+# This Yaml Document has been converted by ESAI Yaml Pipeline Conversion Tool.
+# Please make sure to check all the converted content, it is your team's responsibility to make sure that the pipeline is still valid and functions as expected.
```
Comment (TravisEz13) [inline .pipelines/templates/json.yml:2]:
```suggestion
```

File: .pipelines/templates/insert-nuget-config-azfeed.yml [hunk 44]
```diff
@@ -0,0 +1,35 @@
...
+- pwsh: |
+    $configPath = "${env:NugetConfigDir}/nuget.config"
+    Import-Module ${{ parameters.repoRoot }}/build.psm1 -Force
+    New-NugetConfigFile -NugetFeedUrl $(AzDevOpsFeed) -UserName $(AzDevOpsFeedUserName) -ClearTextPAT $(AzDevOpsFeedPAT2) -FeedName AzDevOpsFeed -Destination "${env:NugetConfigDir}"
+    if(-not (Test-Path $configPath))
+    {
+        throw "nuget.config is not created"
+    }
+    Get-Content $configPath | Write-Verbose -Verbose
+  displayName: 'Add nuget.config for Azure DevOps feed for PSGallery modules'
```
Comment (TravisEz13) [inline .pipelines/templates/insert-nuget-config-azfeed.yml:16]: discuss required changes to internal feed offline.

File: .pipelines/templates/json.yml [hunk 45]
```diff
@@ -0,0 +1,43 @@
...
+  pool:
+    type: windows
+  steps:
+  - checkout: self
+    clean: true
+  - template: /.pipelines/templates/SetVersionVariables.yml@self
+    parameters:
+      ReleaseTagVar: $(ReleaseTagVar)
+      CreateJson: yes
+  - task: AzureFileCopy@4
```
Comment (TravisEz13) [inline .pipelines/templates/json.yml:27]: We should consider separating the build into a build and release jobs

File: .pipelines/PowerShell-Coordinated_Packages-Official.yml (hunk 41 above)
Comment (adityapatwardhan) [reply .pipelines/PowerShell-Coordinated_Packages-Official.yml:95]: Fixed in each job

File: .pipelines/templates/obp-file-signing.yml [hunk 46]
```diff
@@ -0,0 +1,152 @@
+parameters:
+  binPath: '$(ob_outputDirectory)'
+
+steps:
+- pwsh: |
+    $fullSymbolsFolder = '${{ parameters.binPath }}'
```
Comment (TravisEz13) [inline .pipelines/templates/obp-file-signing.yml:6]: We should manually validate Windows file signing

Comment (adityapatwardhan) [reply .pipelines/templates/obp-file-signing.yml:6]: I have validated it. For 1p and 3P signing.

File: .pipelines/templates/json.yml (hunk 45 above)
Comment (adityapatwardhan) [reply .pipelines/templates/json.yml]: This is not used in the pipeline

File: .pipelines/PowerShell-Coordinated_Packages-Official.yml (hunk 39 above)
Comment (adityapatwardhan) [reply .pipelines/PowerShell-Coordinated_Packages-Official.yml]: removed

**Comment by TravisEz13** [comment]
We should have a wrapper for the OneBranch actions like setvariable, but it is not blocking for this PR.

**Review by TravisEz13 — APPROVED** [review]
with comments about future work.

**Review by Moazzem-Hossain-pixel — COMMENTED** [review]
Need support

**Pull request merged**